tests/unittests/sources/azure/test_imds.py

```diff
@@ -481,6 +481,46 @@ def test_non_json_repsonse(
             )
         ]
 
+    def test_logs_unique_errors(
+        self,
+        caplog,
+        mock_requests_session_request,
+        mock_url_helper_time_sleep,
+    ):
+        fake_md = {"foo": {"bar": []}}
+        fake_errors = [
+            fake_http_error_for_code(404),
+            fake_http_error_for_code(410),
```

Comment:
Please add a duplicate fake_error here so we can validate logging is initially silenced for duplicates  resulting in only unique logs and not 1:1 for every fake_error

Reply:
done!